### PR TITLE
Remove references to tekton-chains namespace

### PR DESCRIPTION
**Pull request by ramessesii2**
* The `tekton-chains` namespace is obsolete with the chains deployed with `openshift-pipelines` operator.
* We also don't need the secrets-migrator because with 'chains-secret-admin' sa, we directly create/update the `public-key` in the `openshift-pipelines` namespace now using signing-secrets created by chains.
* remove redundant Role & RB for `chains-secrets-admin` SA `openshift-ingress-operator` namespace - The said Role & RB seems to be needed for infra-deployment's shared-resources component which doesn't seem to exist anymore.

### CODE REVIEW

**Comment by ramessesii2** [comment]
Both of these test are failing due to 
```
task plnsvc-setup has the status "Failed":

- tekton-results-api: Exists, Ready
        - tekton-results-watcher: ..............................tekton-results-watcher not found (timeout)
command terminated with exit code 1
```
/test pipeline-service-test
/test pipeline-service-upgrade-test

**Comment by ramessesii2** [comment]
Hey @sayan-biswas,
In my local testing, I've observed that the current timeout setting of 300 seconds for each deployment check seems to be insufficient. The components deployed by the OpenShift-Pipelines operator can sometimes take longer than expected to become ready, and it's not immediately clear why this occurs. 
For instance, I experienced a timeout specifically with `tekton-triggers-controller`:
```
    - Checking deployment status
        - tekton-pipelines-controller: ........Exists, Ready
        - tekton-triggers-controller: ...........................tekton-triggers-controller not found (timeout)
```

Since these components are part of the first sync-wave in the Argo app, it's essential for them to sync properly. I suspect that these delays could be contributing to frequent failures in `tekton-results-watcher`.
Additionally, there might be an issue with the Watcher deployment itself. Even after waiting for an hour, the Watcher doesn't seem to be up and running, despite the fact that the results API and storage-related objects appear to be in good shape.

**Comment by sayan-biswas** [comment]
> Hey @sayan-biswas, In my local testing, I've observed that the current timeout setting of 200 seconds for readiness check of each deployment seems to be insufficient. The components deployed by the OpenShift-Pipelines operator can sometimes take longer than expected to become ready, and it's not immediately clear why this occurs. For instance, I experienced a timeout specifically with `tekton-triggers-controller`:
> 
> ```
>     - Checking deployment status
>         - tekton-pipelines-controller: ........Exists, Ready
>         - tekton-triggers-controller: ...........................tekton-triggers-controller not found (timeout)
> ```
> 
> Since these components are part of the first sync-wave in the Argo app, it's essential for them to sync properly. I suspect that these delays could be contributing to frequent failures in `tekton-results-watcher`. Additionally, there might be an issue with the Watcher deployment itself. Even after waiting for an hour, the Watcher doesn't seem to be up and running, despite the fact that the results API and storage-related objects appear to be in good shape.

This test is passing in the other PRs.

**Comment by ramessesii2** [comment]
> This test is passing in the other PRs.

Yes, I agree. It's puzzling why Results-watcher ends up being the affected component when other tests or elements fail. It makes it challenging to pinpoint the root cause.

**Comment by ramessesii2** [comment]
A different error at least :
```
task tests-baseline has the status "Failed":

- Image signed: Skip
  - Public key: error: timed out waiting for the condition on pipelineruns/tekton-chains-key-lwdm6
```
Locally, the test passes. Gonna give it another try.
/test pipeline-service-upgrade-test

**Comment by ramessesii2** [comment]
> A different error at least :
> 
> ```
> task tests-baseline has the status "Failed":
> 
> - Image signed: Skip
>   - Public key: error: timed out waiting for the condition on pipelineruns/tekton-chains-key-lwdm6
> ```
> 
> Locally, the test passes. Gonna give it another try. /test pipeline-service-upgrade-test

About failing `pipelines-service-upgrade-test`, I logged into the CI cluster to check what's going on: 
```
➜  k logs -n plnsvc-tests tekton-chains-key-9dvm2-tekton-chains-key-pod

Defaulted container "step-cat" out of: step-cat, prepare (init), place-scripts (init)
2023/09/02 12:29:00 warning: unsuccessful cred copy: ".docker" from "/tekton/creds" to "/": unable to create destination directory: mkdir /.docker: permission denied
++ oc get secret public-key -n tekton-chains -o 'jsonpath={.data.cosign\.pub}'
Error from server (Forbidden): secrets "public-key" is forbidden: User "system:serviceaccount:plnsvc-tests:chains-test" cannot get resource "secrets" in API group "" in the namespace "tekton-chains"
+ PUBLIC_KEY=
```
1. So, apparently [older tests](https://github.com/openshift-pipelines/pipeline-service/blob/main/operator/test/manifests/test/tekton-chains/public-key.yaml#L21) are running from main branch and thereby it tries to query the `tekton-chains` ns for the `public-key`.
While the [changes in tests](https://github.com/openshift-pipelines/pipeline-service/pull/751/files#diff-ce93ae4115b05e980da0be2844b7f100431cef3e4012164becadaf06b941e765R21) in this PR  expects the `public-key` in the `openshift-pipelines` namespace.
2. About `Forbidden` error: This is due to the fact that the tekton chains public-key-viewer role only let's the authenticated users to view the [public-key in the `openshift-pipelines` namespace](https://github.com/openshift-pipelines/pipeline-service/pull/751/files#diff-f87df0dac778d4306014f6db530c98e4c3befd52fc3fcf5b4897131c6bd577deR29-R35). And since the ci is referring the upstream/main branch instead of the PR's ref, the authenticate-user group don't have permission in the `tekton-chains` namespace .
3. Summing up: Although the pipeline-service setup is done via PR's reference in CI, the tests are running from upstream/main it seems and that's why it's failing.
Given, I can't discuss this behaviour with @xinredhat untll he comes back from PTO, I'll refrain from making an effort to fix it myself because it might be the expected way to test a PR; and locally this is tested and verified, I'll appreciate if anyone of you @enarha @sayan-biswas @gabemontero could test it locally and help me merge this PR at your discretion.

File: operator/gitops/argocd/pipeline-service/tekton-chains/chains-secrets-migration.yaml
Comment (gabemontero) [inline operator/gitops/argocd/pipeline-service/tekton-chains/chains-secrets-migration.yaml:113]: @Roming22 - I believe you verified the contents of the secrets this Job migrated when you first introduced the Job, but do you think we need to make sure the secret in openshift-pipelines has not become out of sync since you first checked it, or do you think such a check in unnecessary?

@ramessesii2 fyi

thanks

File: operator/images/cluster-setup/content/bin/install.sh
Comment (gabemontero) [inline operator/images/cluster-setup/content/bin/install.sh:157]: @ramessesii2 @Roming22 do we think we'll ever uncomment these lines and add them back?  If not, we should just delete this commented out check.

**Review by gabemontero — COMMENTED** [review]
One minor question and possible code change suggestion

File: operator/images/cluster-setup/content/bin/install.sh
Comment (ramessesii2) [reply operator/images/cluster-setup/content/bin/install.sh:157]: I'm fine with that. I'll wait for @Roming22's thoughts on this.

File: operator/gitops/argocd/pipeline-service/tekton-chains/chains-secrets-migration.yaml
Comment (ramessesii2) [reply operator/gitops/argocd/pipeline-service/tekton-chains/chains-secrets-migration.yaml:113]: Hi Gabe,

> do you think we need to make sure the secret in openshift-pipelines has not become out of sync

I might have missed what you meant exactly there but here's what I know - the signing-secrets are now directly generated in the openshift-pipelines ns by chains, and we're still [creating/updating the public-key with those secrets](https://github.com/openshift-pipelines/pipeline-service/pull/751/files/16774cd7c8cd4f26b011f72b817584fbe1a46249#diff-31dec81702caa0d2f573347854fd6d2a346d170d139e77d02a19ef6b6d858ac7R78).

I'll wait for anything @Roming22 has to say about it.

File: operator/images/cluster-setup/content/bin/install.sh
Comment (Roming22) [reply operator/images/cluster-setup/content/bin/install.sh:157]: Let's go ahead and delete them.

File: operator/gitops/argocd/pipeline-service/tekton-chains/chains-secrets-migration.yaml
Comment (Roming22) [reply operator/gitops/argocd/pipeline-service/tekton-chains/chains-secrets-migration.yaml:113]: The secret was copied at migration. Since then only the secret in the `openshift-pipelines` has been used, and we have not rotated any secrets since.

Therefore I don't think a check is necessary.

**Review by Roming22 — APPROVED** [review]
LGTM. Clear to be merged once the obsolete commented code has been removed.

**Comment by gabemontero** [comment]
> I've addressed your comments @gabemontero, please merge this PR if it looks good to you. The failing CI is due to a bug, it's being separately worked on by Xin.

Are you talking about @xinredhat 's #758 @ramessesii2 ?  If so, with the back and forth between he and @Roming22 I'm unclear and where that fix might be.

Also, I'm a little unclear on how that addesses the test-baseline failure in the upgrade test, which was chains related:
```
- Image signed: Skip
  - Public key: error: timed out waiting for the condition on pipelineruns/tekton-chains-key-m9h7v
```

now, the pipelines-service-test passed ... so maybe that speaks to your change fixing an issue in the baseline ??

I'm going to defer to @Roming22 for merging this, as he has been more involved in sorting our the CI failures.  Generally, I'm good to go with this, but I don't feel I'm aware of all our current CI error nuances to hit the merge button.

**Comment by ramessesii2** [comment]
Sure Gabe, I'm good with waiting on either @Roming22's discretion to merge it or we fix the CI issue and then merge it.

**Comment by Roming22** [comment]
I'd prefer to fix the CI issue first. But if that is not done by the end of the week, we'll force the merge.

**Comment by Roming22** [comment]
The `cluster-destroy` failed,but everything else was OK, so merging.

**Pull request merged**
